CMakeLists.txt

```diff
@@ -1,141 +1,31 @@
 cmake_minimum_required(VERSION 3.21.4)
 
-cmake_policy(SET CMP0028 NEW)
-cmake_policy(SET CMP0042 NEW)
-cmake_policy(SET CMP0048 NEW)
-cmake_policy(SET CMP0057 NEW)
-cmake_policy(SET CMP0076 NEW)
-
 list(APPEND CMAKE_MODULE_PATH
-     "${CMAKE_CURRENT_LIST_DIR}/cmake"
-     "${CMAKE_CURRENT_LIST_DIR}/cmake/packages"
-     "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/share/modules/cmake")
-
-include(CMConfig)
-include(CMSetupVersion)
-
-if(NOT CMAKE_WORKSPACE_NAME OR NOT ("${CMAKE_WORKSPACE_NAME}" STREQUAL "crypto3"))
-    cm_workspace(crypto3)
-endif()
-
-option(ZK_PLACEHOLDER_PROFILING "Build with placeholder profiling" FALSE)
-option(ZK_PLACEHOLDER_DEBUG "Build with placeholder debug output" FALSE)
-
-if(ZK_PLACEHOLDER_PROFILING)
-    add_definitions(-DZK_PLACEHOLDER_PROFILING_ENABLED)
-endif()
-
-if(ZK_PLACEHOLDER_DEBUG)
-    add_definitions(-DZK_PLACEHOLDER_DEBUG_ENABLED)
-endif()
-
-option(BLUEPRINT_DEBUG "Build with blueprint debug output" FALSE)
-
-if(BLUEPRINT_DEBUG)
-    message(STATUS "BLUEPRINT DEBUG enabled")
-    add_definitions(-DBLUEPRINT_DEBUG_ENABLED)
-endif()
-
-option(BLUEPRINT_TESTS_PROFILING "Build tests with blueprint table profiling" FALSE)
-option(BLUEPRINT_PLACEHOLDER_PROOF_GEN "Build with placeholder proof generation running" FALSE)
-
-if(BLUEPRINT_TESTS_PROFILING)
-    message(STATUS "BLUEPRINT TESTS PROFILING enabled")
-    add_definitions(-DBLUEPRINT_PLONK_PROFILING_ENABLED)
-endif()
-
-if(BLUEPRINT_PLACEHOLDER_PROOF_GEN)
-    message(STATUS "BLUEPRINT PROOF GENERATION enabled")
-    add_definitions(-DBLUEPRINT_PLACEHOLDER_PROOF_GEN_ENABLED)
-endif()
-
-cm_project(blueprint WORKSPACE_NAME ${CMAKE_WORKSPACE_NAME} LANGUAGES C CXX)
+     "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/share/modules/cmake"
+)
 
-include(CMDeploy)
-include(FindPkgConfig)
+project(crypto3_blueprint VERSION 0.1.0 LANGUAGES C CXX)
 
 option(CMAKE_ENABLE_TESTS "Enable tests" FALSE) # used by CMTest module
 option(BUILD_EXAMPLES "Build examples" FALSE)
 
 find_package(crypto3 REQUIRED)
 find_package(Boost REQUIRED COMPONENTS container random filesystem log log_setup program_options thread system)
-if(CMAKE_ENABLE_TESTS)
-    find_package(Boost REQUIRED COMPONENTS unit_test_framework timer)
```

Comment:
Why you're deleting this? We have the same lines in crypto3, we would need to delete it there as well, if there are strong arguments for it
https://github.com/NilFoundation/crypto3/blob/8853748ceffacde6becc9d1ed3f6adacd0f05d29/CMakeLists.txt#L39

Reply:
This `find_package` overwrites `Boost_LIBRARIES` variable that is used below in this file. So in case of enabled tests `blueprint` target would be linked with the wrong set of Boost components. I moved this line into `tests` subdirectory. We need to check `crypto3` as well.